lib/JunkDetection/CXX/CXXJunkDetector.cpp

```diff
@@ -17,26 +17,45 @@
 using namespace mull;
 
 template <typename Visitor>
-static bool isJunkMutation(ASTStorage &storage, MutationPoint *point) {
+static bool
+isJunkMutation(ASTStorage &storage, MutationPoint *point,
+               std::map<MutationPoint *, BeginEndPair> &mutationPointsRanges) {
   auto ast = storage.findAST(point);
   auto location = ast->getLocation(point);
+  auto &sourceManager = ast->getSourceManager();
 
   if (ast->isInSystemHeader(location)) {
     return true;
   }
 
-  VisitorParameters parameters = {.sourceManager = ast->getSourceManager(),
-                                  .sourceLocation = location,
-                                  .astContext = ast->getASTContext()};
+  VisitorParameters parameters = {.sourceManager = sourceManager,
+    .sourceLocation = location,
+    .astContext = ast->getASTContext()};
   Visitor visitor(parameters);
   visitor.TraverseDecl(ast->getASTContext().getTranslationUnitDecl());
 
-  return !visitor.foundMutant();
+  if (!visitor.foundMutant()) {
+    return true;
+  } else {
+
+    clang::SourceRange sourceRange = visitor.getSourceRange();
+    clang::SourceLocation sourceLocationBegin = sourceRange.getBegin();
+    clang::SourceLocation sourceLocationEnd = sourceRange.getEnd();
+
+    unsigned int beginCol =
+      sourceManager.getExpansionColumnNumber(sourceLocationBegin);
+    unsigned int endCol =
+      sourceManager.getExpansionColumnNumber(sourceLocationEnd);
+
+    mutationPointsRanges[point] = std::make_pair(beginCol, endCol);
```

Comment:
As mentioned earlier, this is not enough: what happens when an AST node is scattered across different lines?